### ISSUE TITLE
screen sharing: image badly damaged and flickery

### ISSUE THREAD

**Issue by pschmitt** [hyprwm/Hyprland#4788]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit 301b48b74087cc59753ffa144b215540e6f82831
  ().
Date: 2024-02-18
Tag:

flags: (if any)
```

</details>

### Bug or Regression?

Regression

### Description

did the git bisect already:

<details>
<summary>git bisect log</summary>

```
git bisect start
# status: waiting for both good and bad commits
# bad: [e793f10b8bd1685a2e2bbd7cc90063f12bf380c4] screencopy: fix invalid damage being used for final copy in dma
git bisect bad e793f10b8bd1685a2e2bbd7cc90063f12bf380c4
# status: waiting for good commit(s), bad commit known
# good: [03ebbe18ed8517ee22591eac82cd54322f42cb7d] props: bump ver to 0.34.0
git bisect good 03ebbe18ed8517ee22591eac82cd54322f42cb7d
# good: [4b4bd90b1450cbfc01d9ac429363cc7cecd6be8b] renderer: fixup misaligned fsv1 surfaces with uv
git bisect good 4b4bd90b1450cbfc01d9ac429363cc7cecd6be8b
# good: [f33d73b9cf102d5de7bcc29721ec4f8fc7c0d66d] nix: overlay polish for prev parameter (#4558)
git bisect good f33d73b9cf102d5de7bcc29721ec4f8fc7c0d66d
# good: [a8dae8f5e198327e5de8508d860234089f31e272] socket2: `monitoraddedv2` IPC event for monitor description and id (#4646)
git bisect good a8dae8f5e198327e5de8508d860234089f31e272
# good: [5fc0b772c72ce23f3d31fec31fa3688f7ed1afff] config: update default config for hyprlang migration
git bisect good 5fc0b772c72ce23f3d31fec31fa3688f7ed1afff
# bad: [e4790e3f8e8f3693d924c3d8fc9012e3ab7bbf8f] surface: fix invalid damage tracking in damageSurface
git bisect bad e4790e3f8e8f3693d924c3d8fc9012e3ab7bbf8f
# bad: [301b48b74087cc59753ffa144b215540e6f82831] renderer: fix invalid damage accumulation with invalid buffer_age
git bisect bad 301b48b74087cc59753ffa144b215540e6f82831
# good: [fae47ef462d1d7969afe001070839391ba6dbaa8] config: fix errors in default config
git bisect good fae47ef462d1d7969afe001070839391ba6dbaa8
# first bad commit: [301b48b74087cc59753ffa144b215540e6f82831] renderer: fix invalid damage accumulation with invalid buffer_age
```
</details>

tldr: first bad commit is 301b48b74087cc59753ffa144b215540e6f82831

### How to reproduce

check out commit 301b48b74087cc59753ffa144b215540e6f82831 or later and start a screen share (here for ex: https://www.webrtc-experiment.com/Pluginfree-Screen-Sharing/)

You should see a very weird looking and flickering image of the selected target window.

### Crash reports, logs, images, videos

![2024-02-21T23:18:34+01:00](https://github.com/hyprwm/Hyprland/assets/37886/9363f740-fc69-4f26-b1e2-bd1f0847d355)

**Comment by vaxerski** [comment]
thats old and has been fixed. Update.

**Comment by pschmitt** [comment]
erm this is still very much an issue, even on the current main branch ie ea3fd13e24d7d3a74cf803bef3e6133b5d708d1b

**Comment by vaxerski** [comment]
I have doubts regarding that, though... All seems fine on my end.

**Comment by pschmitt** [comment]
That's sharing an individual window.


![2024-02-21T23:42:46+01:00](https://github.com/hyprwm/Hyprland/assets/37886/ff1a9f09-2c33-460a-a736-37cf790dc02a)

**Comment by pschmitt** [comment]
Region and output sharing works fine. Only window sharing is messed up.

tried with both the latest release of xdph (1.3.1) and the latest main branch

**Comment by vaxerski** [comment]
94aeb06d6ba14d403c46b52d1d2e397acb5906a4 should fix this.

**Comment by pschmitt** [comment]
It does indeed. Danke.
